orchestration/dagster_orchestration/hca_orchestration/tests/contrib/test_argo_workflows.py

```diff
@@ -0,0 +1,219 @@
+                'dataset': 'data'
+            }),
+        ]
+
+        test_mock4 = MagicMock()
+        test_mock4.items = archived_workflows_4
+        test_mock4.metadata._continue = None
+
+        def get_page(list_options_continue=None):
+            if list_options_continue is None:
+                return test_mock
+
+            if list_options_continue == "page2":
+                return test_mock2
+
+            if list_options_continue == "page3":
+                return test_mock3
+
+            if list_options_continue == "page4":
+                return test_mock4
+
+        with patch('argo.workflows.client.ArchivedWorkflowServiceApi.list_archived_workflows',
+                   side_effect=get_page) as mock_list_archived_workflows:
+            list_results = list(client._pull_paginated_results(mock_list_archived_workflows))
+            self.assertEqual(list_results, archived_workflows + archived_workflows_2 + archived_workflows_3 +
+                             archived_workflows_4)
+            self.assertEqual(mock_list_archived_workflows.call_count, 4)
+            mock_list_archived_workflows.assert_has_calls([call(), call(list_options_continue="page2"),
+                                                           call(list_options_continue="page3"),
+                                                           call(list_options_continue="page4")])
+
+    def test_inflate_one_workflow_one_call(self):
+        v1alphaworkflow_mock = mock_argo_workflow('import-hca-total-abcd', 'abc234uid', 'Succeeded', params={
+            'data-repo-name': 'datarepo_dataset1'
+        })
+
+        workflow = extend_workflow(v1alphaworkflow_mock)
+
+        with patch('hca_orchestration.contrib.argo_workflows.ArgoArchivedWorkflowsClient.get_archived_workflow',
+                   return_value=v1alphaworkflow_mock) as mock_get_archived_workflow:
+            workflow.inflate()
+            mock_get_archived_workflow.assert_called_once_with(workflow.metadata.uid)
+            workflow.inflate()
+            mock_get_archived_workflow.assert_called_once_with(workflow.metadata.uid)
+
+    def test_params_dict_workflow_info_dict_proper_values(self):
+        archived_workflows = [
+            mock_argo_workflow('import-hca-total-abcd', 'abc234uid', 'Succeeded', params={
+                'data-repo-name': 'datarepo_dataset1',
+                'name': 'value',
+                'testing': 'data',
+                '1023_3': 'e5Qss'
+            }),
+            mock_argo_workflow('hca_import', 'randomuid', 'Succeeded', params={
+                'dataset_name': 'data'
+            }),
+            mock_argo_workflow('hca_import', 'randomuid', 'Succeeded'),
+        ]
+
+        dict0 = {'data-repo-name': 'datarepo_dataset1',
+                 'name': 'value',
+                 'testing': 'data',
+                 '1023_3': 'e5Qss'}
+        dict1 = {
+            'dataset_name': 'data'
+        }
```

Comment:
if you reorder these statements and pass `dict0` and `dict1` directly into the params argument of `mock_argo_workflow`, i think it'll make this test a bit cleaner. that'll also ensure that you're using the exact same dicts when setting up the workflows as when you're testing their outputs.